### PR TITLE
week-1 tasks of book and user service

### PR DESCRIPTION
**Pull request by sskadam45**

### CODE REVIEW

File: book-service/pom.xml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.springframework.boot</groupId>
+				<artifactId>spring-boot-maven-plugin</artifactId>
+			</plugin>
+		</plugins>
+	</build>
+
+</project>
```
Comment (rupeshpatil761) [inline book-service/pom.xml:105]: Do we really need all the above files and changes for maven?
I think pom.xml is enough in each service.

File: book-service/src/main/java/com/jpop/bookservice/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Autowired
+    BookService bookService;
+
+    @GetMapping("books")
+    List<Book> getAllBooks(){
+        return bookService.findAll();
+    }
+
+    @GetMapping("books/{id}")
```
Comment (rupeshpatil761) [inline book-service/src/main/java/com/jpop/bookservice/controller/BookController.java]: Try to follow versioning.
https://dzone.com/articles/versioning-rest-api-with-spring-boot-and-swagger

File: user-service/src/test/java/com/jpop/userservice/UserServiceApplicationTests.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class UserServiceApplicationTests {
+
+	@Test
+	void contextLoads() {
+	}
+
+}
```
Comment (rupeshpatil761) [inline user-service/src/test/java/com/jpop/userservice/UserServiceApplicationTests.java:13]: Please refer to book service review comments, same goes for user-service

File: book-service/src/main/java/com/jpop/bookservice/service/BookService.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    public Book save(Book book) {
+        return bookRepository.save(book);
+    }
+
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+}
```
Comment (rupeshpatil761) [inline book-service/src/main/java/com/jpop/bookservice/service/BookService.java:41]: You may handle exceptions in a common place.
May be Generic Exception Handler class

File: book-service/src/main/java/com/jpop/bookservice/controller/BookController.java (hunk 2 above)
Comment (sskadam45) [reply book-service/src/main/java/com/jpop/bookservice/controller/BookController.java]: Resolved

**Comment by sskadam45** [comment]
Resolved the review comments. Let me know if anything missing.

**Comment by rupeshpatil761** [comment]
Looks good to me

**Pull request merged**